lib/node/http.js

```diff
@@ -297,13 +297,30 @@ class HTTP extends Server {
     this.get('/mempool', async (req, res) => {
       enforce(this.mempool, 'No mempool available.');
 
-      const hashes = this.mempool.getSnapshot();
-      const result = [];
+      const valid = Validator.fromRequest(req);
+      const details = valid.bool('details', false);
 
-      for (const hash of hashes)
-        result.push(util.revHex(hash));
+      if (details) {
+        const txs = this.mempool.getHistory();
 
-      res.json(200, result);
+        // stream the response
```

Comment:
Here is an example of streaming a potentially large amount of data back to the client. Each transaction in the mempool is serialized individually into a string and written to the response. Still need to take more precise measurements, but I believe that this will be easier on the garbage collector than serializing all of the transactions at once into a large JSON string to send back.